pkg/csi/volume_store.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright © 2021 Alibaba Group Holding Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package csi
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"sync"
+)
+
+type Store interface {
+	AddVolume(volumeID, device string) error
+	DeleteVolume(volumeID string) error
+	GetDevice(volumeID string) string
+}
+
+type volumeStore struct {
+	rwLock             sync.RWMutex
+	volumeDeviceMapper map[string]string
+	dataFilePath       string
+}
+
+func NewVolumeStore(dataFilePath string) (Store, error) {
+	var err error
+	var volumeDeviceMapper map[string]string
+	if volumeDeviceMapper, err = loadVolumeData(dataFilePath); err != nil {
+		if os.IsNotExist(err) {
+			volumeDeviceMapper = make(map[string]string)
+		} else {
+			return nil, err
+		}
+	}
+	return &volumeStore{
+		volumeDeviceMapper: volumeDeviceMapper,
+		dataFilePath:       dataFilePath,
+	}, nil
+}
+
+func (store *volumeStore) AddVolume(volumeID, device string) error {
```

Comment:
do we need to form a rule of what is a valid ID, and the validate it?

Reply:
Volume id is generated by k8s, so we do not need it.

Reply:
ok